### PR TITLE
[Move dashcard to tab] permissions e2e + helper e2e methods

### PR DESCRIPTION
**Pull request by npretto**
epic: https://github.com/metabase/metabase/issues/34367

### Description

- https://github.com/metabase/metabase/commit/8462081fe679786ba234ca067e3f51e0d631c950 adds a e2e to check for the permissions edge case, in which you don't have access to the card but should be able to move the dashcard if you have edit permissions in the dashboard
- https://github.com/metabase/metabase/commit/01b5e992dc05fcde824c3c90824fc5a906121d14 adds a goToTab helper
- https://github.com/metabase/metabase/commit/1b03cf013c9a01e35c7129af7b206f02945ce2f2 adds a helper to move a dashcard to another tab

Note: e2e are probably failing because of https://metaboat.slack.com/archives/C5XHN8GLW/p1697729886291229

### CODE REVIEW

File: e2e/test/scenarios/dashboard/tabs.cy.spec.js [hunk 1]
```diff
@@ -155,25 +156,51 @@ describe("scenarios > dashboard > tabs", () => {
...
+    editDashboard();
+    createNewTab();
+
+    goToTab("Tab 1");
+
+    moveDashCardToTab({ tabName: "Tab 2" });
+
+    saveDashboard();
+
+    getDashboardCards().should("have.length", 0);
```
Comment (npretto) [inline e2e/test/scenarios/dashboard/tabs.cy.spec.js:201]: maybe we could add a reload after the save to make sure everything is actually saved?
but it would slow down the test a bit

**Review by ranquild — APPROVED** [review]
Could we add an assert that the user sees the "No access" message? Currently the test is a bit hard to follow because there is none
